plugin-hrm-form/src/utils/sharedState.js

```
@@ -66,3 +70,72 @@ export const loadFormSharedState = async task => {
     return null;
   }
 };
+
+/**
+ * This function creates an object with all parseable attributes from the original Twilio Task.
+ *
+ * The twilio task by itself cannot be passed directly to be store at the Shared State.
+ * This happens because, internally, it tries to parse the object but it fails because of circular dependencies.
+ *
+ * @param {*} task Twilio Task
+ * @returns Parseable copy of the Twilio Task
+ */
+const copyTask = task => {
+  const taskToReturn = {};
+  Object.keys(task._task).forEach(key => {
+    const value = task[key];
+    if (['object', 'string', 'number'].includes(typeof value)) {
+      taskToReturn[key] = task[key];
+    }
+  });
+
+  return taskToReturn;
+};
+
+/**
+ * This function receives an Error and returns an error object with the following attributes:
+ * - message: the Error.message
+ * - stack: the original stack trace (that's optionally store at Error.cause)
+ * @param {*} error an instance of Error
+ * @returns error object to be stored at the Sync Client
+ */
+const copyError = error => ({
+  message: error.message,
+  ...(error.cause && { stack: error.cause }),
+});
+
+/**
+ * Saves a pending contact into the Sync Client
+ * @param {import("@twilio/flex-ui").ITask} task task used to save the contact
+ * @param {*} payload payload used to save the contact
+ * @param {*} error error returned when trying to save contact to external backend
+ */
+export const savePendingContactToSharedState = async (task, payload, error) => {
+  const { featureFlags, sharedStateClient, strings } = getConfig();
+
+  if (!featureFlags.enable_dual_write) return null;
+  if (!task || !payload) return null;
+
+  try {
+    if (!isSharedStateClientConnected(sharedStateClient)) {
```

Same here, should we be swallowing exceptions in low level client / utility functions? Whilst this error handling might be appropriate for the current use case, it's possible we might start calling it in other scenarios that want to handle the error differently?